### PR TITLE
fix(cdn): rework lru writer to allows concurrent write

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14841.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14841.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✘  
  * Sonar ✔

File: engine/cdn/redis/writer.go [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
-		n++
-		if charact == "\n" {
-			if err := w.ReadWrite.add(w.CurrentScore, currentLine); err != nil {
-				return 0, err
-			}
-			w.CurrentScore++
-			currentLine = ""
+	// Save all lines except the last one
+	for i := 0; i < len(bufferSplitted); i++ {
+		// For last part we add the bytes to the current buffer as it can be a partial line
```
Comment (fsamin) [inline engine/cdn/redis/writer.go:23]: If we keep a partial line in the current buffer, how we are handling it ?
It looks like we keep it for the next `Write` call, but what if it never happens ?
Maybe we should handle it in a `Close()`method...

**Review by sguiheux — APPROVED** [review]

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14851.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by yesnault — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

**Pull request merged**